packages/web-components/src/components/rux-slider/rux-slider.tsx

```diff
@@ -157,46 +178,66 @@ export class RuxSlider implements FormFieldInterface {
-        if (!this.min && this.min != 0) {
-            this.min = 0
-        }
-        //If max is not a number, change it to 100
-        if (!this.max && this.max != 0) {
-            this.max = 100
-        }
-        // If value is not a number, change it to default.
-        if (!this.value && this.value != 0) {
-            this.value = (this.max - this.min) / 2 + this.min
-        }
-        //If step is not a number, change it to 1
-        if (!this.step) {
-            this.step = 1
-        }
-        //Min can't be >= max
-        if (this.min >= this.max) {
-            this.min = this.max - this.step
-        }
-        // If min is given and is greater than value, then set value to the min.
-        if (this.value < this.min) {
-            this.value = this.min
-        }
-        //If max is given and is less than value, set value to max
-        if (this.max < this.value) {
-            this.value = this.max
-        }
-
         this._setValuePercent()
     }
-    //Sets the --slider-value-percent CSS var
+
+    //Sets the --slider-value-percent CSS var. Also contor
     private _setValuePercent() {
-        const dif = ((this.value! - this.min!) / (this.max! - this.min!)) * 100
-        this.el.style.setProperty('--_slider-value-percent', `${dif}%`)
+        //if minVal is being used, we're in dual range mode.
+        if (this.minVal !== undefined) {
+            if (this.minVal > this.value) {
+                this.el.style.setProperty(
+                    '--_slider-value-percent',
+                    `${this.minVal}%`
+                )
+                this.el.style.setProperty(
+                    '--_start-value-percent',
+                    `${this.value}%`
+                )
+                //If end < start, no need to swap
+            } else {
+                this.el.style.setProperty(
+                    '--_start-value-percent',
+                    `${this.minVal}%`
+                )
+                this.el.style.setProperty(
+                    '--_slider-value-percent',
+                    `${this.value}%`
+                )
+            }
+            //if not in dual slider
+        } else {
+            const dif =
+                ((this.value! - this.min!) / (this.max! - this.min!)) * 100
+            this.el.style.setProperty('--_slider-value-percent', `${dif}%`)
+        }
     }
 
     private _onInput(e: Event) {
         const target = e.target as HTMLInputElement
-        this.value = parseFloat(target.value)
+        if (this.value !== undefined) {
+            this.value = parseFloat(target.value)
+            if (this.value <= this.minVal! && this.strict) {
+                this.value = this.minVal!
+                // this.value = this.maxVal!
+                //? Still not entirely sure why we have to do this.
+                target.value = this.value.toString()
+            }
+        } else {
+            this.value = parseFloat(target.value)
```

Comment:
Do you need this else clause? It looks like you're setting `this.value` as a float regardless. Can you bring back line 199 where you set it before hitting the first if statement?

Reply:
nope, that's extra